include/boost/gil/image_processing/threshold.hpp

```diff
@@ -179,6 +181,132 @@ void threshold_truncate
     }
 }
 
+namespace detail{
+
+template <typename SrcView, typename DstView>
+void otsu_impl(SrcView const& src_view, DstView const& dst_view, threshold_direction direction)
+{
+    //deciding output channel type and creating functor
+    typedef typename channel_type<SrcView>::type source_channel_t;
+    typedef typename channel_type<DstView>::type result_channel_t;
+
+    std::array<std::size_t, 256> histData{};
```

Comment:
@miralshah365 GIL, as well as majority of Boost libraries, follow the C++ standard library naming conventions where the names of variables, function parameters, class data members are all lowercase, with underscores between words.

```
- histData
+ histogram
```

`histData`